consensus/errors.go

```diff
@@ -0,0 +1,44 @@
+package consensus
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrNilMessage                    = errors.New("message is nil")
+	ErrPeerStateHeightRegression     = errors.New("error peer state height regression")
+	ErrPeerStateInvalidStartTime     = errors.New("error peer state invalid startTime")
+	ErrCommitQuorumNotMet            = errors.New("extended commit does not have +2/3 majority")
+	ErrNilPrivValidator              = errors.New("entered createProposalBlock with privValidator being nil")
+	ErrProposalWithoutPreviousCommit = errors.New("propose step; cannot propose anything without commit for the previous block")
+)
+
+// Consensus sentinel errors
+var (
+	ErrInvalidProposalSignature   = errors.New("error invalid proposal signature")
+	ErrInvalidProposalPOLRound    = errors.New("error invalid proposal POL round")
+	ErrAddingVote                 = errors.New("error adding vote")
+	ErrSignatureFoundInPastBlocks = errors.New("found signature from the same key")
+	errPubKeyIsNotSet             = errors.New("pubkey is not set. Look for \"Can't get private validator pubkey\" errors")
```

Comment:
Since this is returned by some functions, should it not be exported?

Reply:
it is only used in `consensus` package. however, same can be said about other errors under `Consensus sentinel errors`. these errors were introduced prior to the time I started working on exporting errors, so I only moved them to `errors.go` w/o changing anything else. personally, I also think it is worth to export it